cores/esp32/esp32-hal-adc.c

```diff
@@ -46,7 +46,11 @@ static uint8_t __analogVRefPin = 0;
 #endif
 
 static uint8_t __analogAttenuation = 3;//11db
+#if CONFIG_IDF_TARGET_ESP32
 static uint8_t __analogWidth = 3;//12 bits
+#elif CONFIG_IDF_TARGET_ESP32S2
```

Comment:
This should be listed first and have an `#else` with the existing 12bit analog value.  With the proposed change this will fail to compile on the ESP32-C3.